solidity/src/Router7683.sol

```diff
@@ -39,19 +42,34 @@ contract Router7683 is GasRouter, Base7683 {
         _MailboxClient_initialize(_customHook, _interchainSecurityModule, _owner);
     }
 
-    // ============ External Functions ============
-
-    // TODO - implement interchain settlement functions
-
     // ============ Internal Functions ============
 
-    function _handle(uint32 _origin, bytes32 _sender, bytes calldata _message) internal virtual override {
-        // TODO - handle settlement
+    function _handle(uint32 _origin, bytes32, bytes calldata _message) internal virtual override {
+        (bool _settle, bytes32[] memory _orderIds, bytes32[] memory _receivers) = Router7683Message.decode(_message);
+
+        if (_settle) {
+            for (uint i = 0; i < _orderIds.length; i++) {
+                _settleOrder(_orderIds[i], _receivers[i], _origin);
+            }
+        } else {
+            for (uint i = 0; i < _orderIds.length; i++) {
+                _refundOrder(_orderIds[i], _origin);
+            }
+        }
+    }
+
+    function _handleSettlement(bytes32[] memory _orderIds, bytes32[] memory _receivers) internal virtual override {
+        _dispatchMessage(true, _orderIds, _receivers);
```

Comment:
Can't say I love this flag in the internal methods. IMO it should have explicit `_dispatchSettlement` or `dispatchRefund` methods and inside those you just call `_GasRouter_dispatch`. I know its a bit more duplication, but imo much more clear to read.